### PR TITLE
More DateTime renames

### LINKED ISSUE #5840: Figure out how to implement DateTimeNames::try_with_pattern
Currently I have an experimental function signature that takes `&mut self`, even though it should not ever mutate. This is because internally it calls `load_for_pattern` with an empty data provider. I don't want to duplicate the code paths.

What should we do?

### PR DESCRIPTION
**Pull request by sffc**
Depends on #5836
Fixes #5840

### CODE REVIEW

**Comment by sffc** [comment]
I think this is reviewable commit by commit if you prefer

**Comment by sffc** [comment]
@Manishearth Please verify that you are reasonably happy with the `EmptyDataProvider` solution to #5840

File: components/datetime/src/pattern/names.rs [hunk 1]
```diff
@@ -124,6 +136,13 @@ size_test!(
 ///         (45, 50, Part::ERROR), // +0000
 ///     ]
 /// );
+///
+/// // To make the error occur sooner, one can use an EmptyDataProvider:
+/// let empty = EmptyDataProvider::new();
+/// assert!(matches!(
+///     names.load_for_pattern(&empty, &pattern),
+///     Err(PatternLoadError::Data(_, Field { symbol: FieldSymbol::Weekday(_), .. })),
+/// ));
```
Comment (sffc) [inline components/datetime/src/pattern/names.rs:145]: Highlighting the new doc example here, the one that resolves #5840

**Review by Manishearth — APPROVED** [review]

File: components/datetime/src/pattern/pattern.rs [hunk 2]
```diff
@@ -13,14 +13,14 @@ size_test!(DateTimePattern, date_time_pattern_size, 32);
 
 /// A pattern for formatting a datetime in a calendar.
 ///
-/// [`DateTimePattern`] forgoes most internationalization functionality of the datetime crate.
+/// ❗ This type forgoes most internationalization functionality of the datetime crate.
```
Comment (Manishearth) [inline components/datetime/src/pattern/pattern.rs:16]: praise: good call

**Pull request merged**